### PR TITLE
Clean redeem process

### PR DESCRIPTION
**Pull request by clearloop**
## Steps

+ [x] Merge the config of Ethereum listener
+ [x] Delay the relay process after scanning Ethereum transactions
+ [x] Clean the Ethereum listener
+ [x] Remove all verbose commands
+ [x] Write a new CLI entry
+ [x] Clean deps
+ [x] test api

### CODE REVIEW

File: src/listener/relay.ts [hunk 1]
```diff
@@ -1,68 +1,59 @@
...
-                log.trace(`\t\t${event.meta.documentation.toString()}`);
-                const lastConfirmed = await api.lastConfirm();
-                queue.filter((tx) => tx.relayedBlock === lastConfirmed).forEach(async (tx) => {
-                    await api.redeem(tx.ty, await shadow.getReceipt(tx.tx, lastConfirmed));
-                });
+        /// Check last confirm
+        lastConfirmed = await api.lastConfirm();
 
-                queue = queue.filter((tx) => tx.relayedBlock !== lastConfirmed);
+        // Refresh target
```
Comment (wuminzhe) [inline src/listener/relay.ts:19]: maybe this is a easy way:
```
target = max_block_number_of(queue)
if last_confirmed >= target { 
	continue
}
```

Comment (clearloop) [reply src/listener/relay.ts:19]: Will modify to 

```js
const target = Math.max(
  lastConfirmed, 
  queue.sort((p, q) => q.blockNumber - p.blockNumber)[0],
) + 7;
```

File: src/listener/relay.ts [hunk 2]
```diff
@@ -73,3 +64,78 @@ export function relay(api: API, shadow: ShadowAPI, queue: ITx[]) {
...
+    phase: any,
+    api: API,
+    shadow: ShadowAPI,
+    queue: ITx[],
+    lastConfirmed: number,
+) {
+    log.trace(`\t${event.section}:${event.method}:: (phase=${phase.toString()})`);
+    log.trace(`\t\t${event.meta.documentation.toString()}`);
+    for (const tx of queue.filter((ftx) => ftx.blockNumber < lastConfirmed)) {
+        lastConfirmed = await api.lastConfirm();
```
Comment (wuminzhe) [inline src/listener/relay.ts:85]: Why get the `lastConfirmed` every time here? If you want to get latest `lastConfirmed`, I think you should filter the queue every time too.

File: src/listener/relay.ts [hunk 3]
```diff
@@ -73,3 +64,78 @@ export function relay(api: API, shadow: ShadowAPI, queue: ITx[]) {
...
+        await api.redeem(tx.ty, await shadow.getReceipt(tx.tx, lastConfirmed));
+        await delay(20000);
+    };
+
+    queue = queue.filter((tx) => tx.blockNumber > lastConfirmed);
```
Comment (wuminzhe) [inline src/listener/relay.ts:90]: I feel there is a small problem here. Off
the `lastConfirmed` is not equal to the one from `for (const tx of queue.filter((ftx) => ftx.blockNumber < lastConfirmed))`, so maybe some tx will be missed this turn.

Comment (clearloop) [reply src/listener/relay.ts:90]: Good catch, should not reset `lastConfirmed` every time, Txs will be missed in this last filter.

This problem related to the [comment](https://github.com/darwinia-network/dj/pull/79/files/991cae74c78c723df9889fc8a50e2641407ef86d#r486844081) above.

File: src/listener/relay.ts (hunk 2 above)
Comment (clearloop) [reply src/listener/relay.ts:85]: Should create a temporary constant like `curLastConfirmed` instead re-use the `lastConfirmed` to handle the redeem ex.

**Review by wuminzhe — APPROVED** [review]

**Pull request merged**